pkg/kubelet/kubelet.go

```diff
@@ -491,7 +491,12 @@ func NewMainKubelet(kubeCfg *kubeletconfiginternal.KubeletConfiguration,
 			clusterDNS = append(clusterDNS, ip)
 		}
 	}
-	httpClient := &http.Client{}
+
+	transport := http.DefaultTransport.(*http.Transport).Clone()
+	transport.TLSClientConfig = &tls.Config{InsecureSkipVerify: true}
```

Comment:
What's the purpose of enabling TLS and skip the certificate verifications?

Reply:
Originally, https was not supported. This is to support https and skip TLS, like Prober